mesonbuild/dependencies/ui.py

```diff
@@ -414,6 +411,26 @@ def get_exe_args(self, compiler):
     def get_private_includes(self, mod_inc_dir, module):
         return tuple()
 
+    def _get_windows_libfile(self, module, prefix = '', suffix = ''):
+        is_debug = self.env.coredata.get_builtin_option('buildtype') == 'debug'
+        dbg = 'd' if is_debug else ''
+        base_name = prefix + module + dbg + suffix
+        libfile = os.path.join(self.libdir, base_name + '.lib')
+        if os.path.isfile(libfile):
+            return libfile
+        # MinGW
+        libfile = os.path.join(self.libdir, 'lib' + base_name + '.a')
+        if os.path.isfile(libfile):
+            return libfile
+        return None
+
+    def get_gui_app_link_args(self):
```

Comment:
Ideally you should implement this in the base `Dependency` class in base.py, and have it just `return []`. Then this will merely override it. That way you can also remove the `if hasattr(dep, 'get_gui_app_link_args')` check above.

Reply:
Thanks for the feedback. BTW, can we get an id of compiler in this function? With current code if compiler is MinGW, it calls os.path.isfile twice for each module, first for .lib file, then for .a file. It would better to actually check for compiler and search for appropriate file.